### PR TITLE
Add action to setup proxy and output sonar opts

### PR DESCRIPTION
**Pull request by bgzzz**
Add proxy code and docker definition
Add gha to run proxy and output sonar opts

### CODE REVIEW

**Review by franciscoalves — APPROVED** [review]
🥇

File: .github/workflows/master.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+        - name: Auth Docker GCR
+          uses: tradeshift/actions-docker-gcr/auth@master
+          with:
+            gcr-service-account-key: ${{ secrets.GCLOUD_SERVICE_ACCOUNT_KEY }}
+
+        - name: Build and push master
+          run: |
+            docker build -f script-proxy-opts/Dockerfile.proxy -t eu.gcr.io/tradeshift-base/sonar-proxy:master .
+            docker push eu.gcr.io/tradeshift-base/sonar-proxy:master
```
Comment (blacksails) [inline .github/workflows/master.yaml:24]: Doesn't actions-docker-gcr support this? If not I guess it should?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,58 @@
...
+          with:
+            ca-cert: ${{ secrets.MTLS_CACERT }}
+            cert: ${{ secrets.MTLS_CERT }}
+            key: ${{ secrets.MTLS_KEY }}
+            sonar-token: ${{ secrets.SONAR_TOKEN }}
+            sonar-host: "https://sonar.host"
+
+        - name SonarQube maven scan
+          run: |
+            mvn -B sonar:sonar ${{ steps.sonar-opts.output.opts }}
```
Comment (blacksails) [inline README.md:55]: I forgot, why is it that sonar needs docker to be configured? Or is it maven that needs that?

**Review by blacksails — APPROVED** [review]
LGTM

File: README.md (hunk 2 above)
Comment (bgzzz) [reply README.md:55]: we need to pull proxy

File: .github/workflows/master.yaml (hunk 1 above)
Comment (bgzzz) [reply .github/workflows/master.yaml]: latest not master

**Pull request merged**